### PR TITLE
smart catch while farming

### PR DESCRIPTION
**Pull request by mohitmun**
Short Description: 
While Farming, Keep single copy of pokemon with highest CP,  and exchange other for candy

### CODE REVIEW

File: cell_workers/pokemon_catch_worker.py [hunk 1]
```diff
@@ -105,6 +106,19 @@ def work(self):
                                 if status is 3:
                                     print('[x] Oh no! ' + str(pokemon_name) + ' vanished! :(')
                                 if status is 1:
+                                    self.api.get_inventory()
```
Comment (solderzzc) [inline cell_workers/pokemon_catch_worker.py]: I remember there's another function to get_inventory(), right?

**Comment by douglascamata** [comment]
you sure this works?

**Comment by mohitmun** [comment]
yeah It works for me

File: cell_workers/pokemon_catch_worker.py (hunk 1 above)
Comment (mohitmun) [inline cell_workers/pokemon_catch_worker.py]: there are multiple calls to 'count_pokemon_inventory' which calls get_inventory. so I moved this api call up and use response to do other operations

Comment (douglascamata) [inline cell_workers/pokemon_catch_worker.py]: smart!

**Comment by douglascamata** [comment]
@mohitmun please add information about this feature in the README in the command-line description.

File: cell_workers/pokemon_catch_worker.py (hunk 1 above)
Comment (mohitmun) [inline cell_workers/pokemon_catch_worker.py]: exachange is super fast. get_inventory moves lots of data on higher level

Comment (mohitmun) [inline cell_workers/pokemon_catch_worker.py]: we can remove 'count_pokemon_inventory' and use directly response

**Comment by mohitmun** [comment]
Yeah sure!

File: cell_workers/pokemon_catch_worker.py (hunk 1 above)
Comment (douglascamata) [inline cell_workers/pokemon_catch_worker.py]: @mohitmun leave this for another pull request, because that thing is big and dangerous.

File: cell_workers/pokemon_catch_worker.py [hunk 2]
```diff
@@ -105,6 +106,19 @@ def work(self):
...
+                                        prev_id, prev_cp = (0,0)
+                                        for id_cp in id_cp_tuples:
+                                            next_id,next_cp = id_cp
+                                            if next_cp < prev_cp:
+                                                self.transfer_pokemon(next_id)
+                                            elif prev_id != 0:
+                                                self.transfer_pokemon(prev_id)
+                                            prev_id,prev_cp = id_cp
+
                                     if cp < self.config.cp:
```
Comment (douglascamata) [inline cell_workers/pokemon_catch_worker.py]: @mohitmun I still see a possibility here of code getting inside `smart_catch` if and inside this one. So I have a new idea! Instead of adding a new option for smart catching, why not use `-c` for this? You can make `-c` accept a string and then you check if the string is `smart` to use smart catch. If the string isn't `smart`, then you try to convert it to int.

Comment (mohitmun) [inline cell_workers/pokemon_catch_worker.py]: make sense

**Comment by douglascamata** [comment]
@mohitmun I was eager for your update. Gonna review!

File: cell_workers/pokemon_catch_worker.py [hunk 3]
```diff
@@ -105,7 +106,24 @@ def work(self):
...
+                                        id_cp_tuples = self.get_id_cp_tuples_for_pokemonid(pokemon['pokemon_data']['pokemon_id'],response_dict)
+                                        prev_id, prev_cp = (0,0)
+                                        for id_cp in id_cp_tuples:
+                                            next_id,next_cp = id_cp
+                                            if next_cp < prev_cp:
+                                                self.transfer_pokemon(next_id)
+                                            elif prev_id != 0:
+                                                self.transfer_pokemon(prev_id)
+                                            prev_id,prev_cp = id_cp
+                                    try:
```
Comment (douglascamata) [inline cell_workers/pokemon_catch_worker.py]: @mohitmun please add an `else` clause to the `if self.config.cp == "smart"` if and move the `try` block inside it.

**Comment by solderzzc** [comment]
I little bit confusing..... merged....

**Comment by mohitmun** [comment]
New to python :P

**Comment by mohitmun** [comment]
@solderzzc can you please try it. It works for me, not confident
